backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt

```diff
@@ -316,6 +318,9 @@ internal val dependenciesLowerPhase = SameTypeNamedPhaseWrapper(
                         }
                 input.files += files
 
+                if (context.config.produce == CompilerOutputKind.PROGRAM)
```

Comment:
Adding entry point seems unrelated to lowering dependencies. Maybe extract explicit phase for this?

Reply:
Well, not sure if it shall a phase on its own. WDYT?

Reply:
What do you mean?